src/components/weekSwitch/weekSwitch.tsx

```diff
@@ -1,10 +1,10 @@
 import OptionMultipleToggler from "../../components/optionMultipleToggler";
 import { useWeekContext } from "../../common/context/weekContext";
 
-const WeekSwitch = ({ type }) => {
-  const { setCurrentWeek, currentWeek } = useWeekContext();
+const WeekSwitch = ({ type }: { type: "weeks" | "semesters" }) => {
```

Comment:
Please extract this type into separate WeekSwitchProps interface and put it above.